### PR TITLE
Add some unit tests for osd .

### PR DESCRIPTION
**Pull request by zhangmengqianyw**
Signed-off-by: Zhang Meng Qian <zhangmengqian_yewu@cmss.chinamobile.com>

### CODE REVIEW

**Comment by epuertat** [comment]
Welcome @zhangmengqianyw as a Ceph contributor!

Could you please squash all your commits into a single one? Additionally [the commit title](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#commit-title) should start with "mgr/dashboard: <short description>".

Thank you very much for your contribution!

File: src/pybind/mgr/dashboard/tests/test_osd.py [hunk 1]
```diff
@@ -313,7 +335,34 @@ def test_osd_mark_all_actions(self, instance):
...
         action_list = ['OUT', 'IN', 'DOWN']
+
         for action in action_list:
             data = {'action': action}
             self._task_put('/api/osd/1/mark', data)
             self.assertStatus(200)
+
+        # invalid mark
+        self._task_put('/api/osd/1/mark', {'action': 'OTHER'})
+        self.assertStatus(200)
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/tests/test_osd.py:346]:
```suggestion
        # invalid mark
        instance.reset_mock()
        with self.assertLogs(level='ERROR') as cm:
            self._task_put('/api/osd/1/mark', {'action': 'OTHER'})
            instance.send_command.assert_not_called()
            self.assertIn('Invalid OSD mark action', cm.output[0])
            self.assertStatus(200)
            self.assertJsonBody(None)
```
Make it more meaningful by checking the error log and `CephService.send_command` is not called.

File: src/pybind/mgr/dashboard/tests/test_osd.py [hunk 2]
```diff
@@ -313,7 +335,34 @@ def test_osd_mark_all_actions(self, instance):
...
+
+        self._task_post('/api/osd/1/purge', {'svc_id': 1})
+        self.assertStatus(200)
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/tests/test_osd.py:349]:
```suggestion
        instance.send_command.assert_called_once_with('mon', 'osd purge-actual', id=1,
                                                      yes_i_really_mean_it=True)
        self.assertStatus(200)
        self.assertJsonBody(None)
```

File: src/pybind/mgr/dashboard/tests/test_osd.py [hunk 3]
```diff
@@ -313,7 +335,34 @@ def test_osd_mark_all_actions(self, instance):
...
+
+    @mock.patch('dashboard.controllers.osd.CephService')
+    def test_destory_osd(self, instance):
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/tests/test_osd.py:352]:
```suggestion
    def test_reweight_osd(self, instance):
```

File: src/pybind/mgr/dashboard/tests/test_osd.py [hunk 4]
```diff
@@ -313,7 +335,34 @@ def test_osd_mark_all_actions(self, instance):
...
+        data = {
+            'method': 'bare',
+            'data': sample_data,
+            'tracking_id': 'bare-1'
+        }
+        self._task_post('/api/osd', data)
+        self._task_put('/api/osd/1/mark', {'action': 'DOWN'})
+        self.assertStatus(200)
+        self._task_post('/api/osd/1/reweight', {'weight': '1'})
+        self.assertStatus(200)
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/tests/test_osd.py:368]:
```suggestion
        instance.send_command.assert_called_with('mon', 'osd reweight', id=1, weight=1.0)
        self.assertStatus(200)
```

**Review by alfonsomthd — CHANGES_REQUESTED** [review]
Thanks for your contribution! I left some suggestions.

**Pull request closed without merging**